validation/src/main/scala/hmda/validation/engine/ts/TsValidationEngine.scala

```diff
@@ -1,59 +1,57 @@
 package hmda.validation.engine.ts
 
 import hmda.model.fi.ts.TransmittalSheet
-import hmda.validation.api.TsValidationApi
-import hmda.validation.dsl.{ Failure, Result, Success }
+import hmda.validation.api.ValidationApi
+import hmda.validation.api.ts.TsValidationApi
 import hmda.validation.engine.ValidationError
 import hmda.validation.rules.ts.syntactical._
-
 import scala.concurrent.{ ExecutionContext, Future }
-import scalaz.Scalaz._
 import scalaz._
+import scalaz.Scalaz._
 
-trait TsValidationEngine extends TsValidationApi {
+trait TsValidationEngine extends ValidationApi with TsValidationApi {
```

Comment:
I'm curious: How do you decide when to use `class` vs `object` vs `trait`? Does the fact that `TsValidationEngine` is a `trait` here mean that another class will eventually extend it?